### PR TITLE
added an endpoint for displaying top rated movies based on our user s…

### PR DESCRIPTION
**Pull request by GeraltWitcher1**
image

### CODE REVIEW

File: BestMovies.Api/Repositories/Impl/StatisticsRepository.cs [hunk 1]
```diff
@@ -32,6 +33,35 @@ public async Task<decimal> GetAverageRatingOfMovies(IEnumerable<int> movieIds)
         return (decimal) Math.Round(average, 2, MidpointRounding.AwayFromZero);
     }
     
+    public async Task<IEnumerable<int>> GetTopRatedMovieIds(int count)
+    {
+        const int minimumReviewCount = 0;
```
Comment (Ion-Caus) [inline BestMovies.Api/Repositories/Impl/StatisticsRepository.cs:38]: why zero? and you can make it a const in the whole repo

File: BestMovies.Api/Repositories/Impl/StatisticsRepository.cs [hunk 2]
```diff
@@ -32,6 +33,35 @@ public async Task<decimal> GetAverageRatingOfMovies(IEnumerable<int> movieIds)
...
+        command.CommandText = @"
+            SELECT TOP (@NumberOfMovies) [MovieId]
+            FROM [Reviews]
+            GROUP BY [MovieId]
+            HAVING COUNT(*) >= @MinimumReviewCount
+            ORDER BY AVG(CAST(Rating AS FLOAT)) DESC;
+        ";
+        
+        command.Parameters.AddWithValue("@NumberOfMovies", count);
+        command.Parameters.AddWithValue("@@MinimumReviewCount", minimumReviewCount);
```
Comment (Ion-Caus) [inline BestMovies.Api/Repositories/Impl/StatisticsRepository.cs:51]: does it need double `@`?

**Review by Ion-Caus — CHANGES_REQUESTED** [review]

File: BestMovies.Api/Repositories/Impl/StatisticsRepository.cs (hunk 2 above)
Comment (GeraltWitcher1) [reply BestMovies.Api/Repositories/Impl/StatisticsRepository.cs:51]: no, good catch

File: BestMovies.Api/Repositories/Impl/StatisticsRepository.cs (hunk 1 above)
Comment (GeraltWitcher1) [reply BestMovies.Api/Repositories/Impl/StatisticsRepository.cs:38]: Also good catch. How can I make it a constant for repo?

**Review by Ion-Caus — APPROVED** [review]

**Pull request merged**